stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -123,6 +123,9 @@ factorize(D::Diagonal) = D
 real(D::Diagonal) = Diagonal(real(D.diag))
 imag(D::Diagonal) = Diagonal(imag(D.diag))
 
+iszero(D::Diagonal) = all(iszero, D.diag)
+iszero(D::Diagonal{<:Number}) = iszero(D.diag)
```

Comment:
This method is redundant with the one on the line above, no?

Reply:
Ah, thats right. Thanks.